### ISSUE TITLE
Implement grouping options for money types

### ISSUE THREAD

**Issue by kgodey** [mathesar-foundation/mathesar#419]
## Problem
We need to ensure that records that include columns of all Money types support range based grouping just like Number types via the API. Please see #384 for the implementation details. 

We should be able to reuse the implementation for Number types here.

This involves:
- Implementing the grouping in the backend
- Updating the `/api/v0/databases/<id>/types/` endpoint to store available grouping on this type

## Additional context
- Please see these PRs for code related to grouping:
  - #315
  - #353
- We're using our fork of `sqlalchemy-filters` to provide filtering. See: https://github.com/centerofci/sqlalchemy-filters.
- This issue is blocked by #384 
- This issue is also blocked by: 
  - #417
  - #420
- #557 provides some context on storing groups.

**Comment by Aditramesh** [comment]
@kgodey can i work on this?

**Comment by kgodey** [comment]
@Aditramesh This is a really old issue and I'm not sure how valid it is. Please check to see if money data types support all the same grouping types as number data types – if they don't, then you can work on it.

**Comment by Aditramesh** [comment]
@kgodey i can try doing that but if there is an issue which has higher priority and if you think I can solve it ,can you assign that to me ?

**Comment by Aditramesh** [comment]
Understood,then I will try to work on this ,thanks @kgodey

**Comment by Anish9901** [comment]
Unassigned due to inactivity.

**Comment by Aditramesh** [comment]
@Anish9901 i couldnt work much on this issue in the last two weeks but i have made some progress and i want to continue working on it,could you reassign this ?

**Comment by Anish9901** [comment]
Please create a PR @Aditramesh and I'll reassign the issue to you.

**Comment by kgodey** [comment]
Closing this, too old, requirements are out of date. We can create a new issue if we need this functionality in the future.
